### PR TITLE
move tests with external servers to outerloop

### PR DESCRIPTION
**Pull request by wfurt**
Update concurency tests to:
- move GetAsync_MaxLimited_ConcurrentCallsStillSucceed to outerloop as it uses external Azure serverts
- remove unnecesary craft from Set_ValidValues_Success. It seems like the comment about curl is no longer true.

### CODE REVIEW

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.MaxConnectionsPerServer.cs [hunk 1]
```diff
@@ -47,19 +47,12 @@ public void Set_InvalidValues_Throws(int invalidValue)
         [InlineData(2)]
         [InlineData(int.MaxValue)]
         [InlineData(int.MaxValue - 1)]
+        [OuterLoop("Uses external servers")]
```
Comment (stephentoub) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.MaxConnectionsPerServer.cs]: Where?  It doesn't make any requests at all, does it?

Comment (wfurt) [reply src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.MaxConnectionsPerServer.cs]: That was mistake. This was really experiment to see if the notes about curl are still valid.

**Comment by wfurt** [comment]
test failures are unrelated e.g. none is in HTTP tests.

**Pull request merged**